app/models/hosts_storage.rb

```diff
@@ -1,2 +1,4 @@
 class HostsStorage < ActiveRecord::Base
```

Comment:
The model might need a `self.primary_key = nil`...then at least update_attributes won't fail, I think.